src/pkg/cli/client/byoc/gcp/byoc.go

```diff
@@ -396,6 +398,34 @@ func (b *ByocGcp) Preview(ctx context.Context, req *defangv1.DeployRequest) (*de
 	return b.deploy(ctx, req, "preview")
 }
 
+func (b *ByocGcp) GetDeploymentStatus(ctx context.Context) error {
+	client, err := run.NewExecutionsClient(ctx)
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+
+	execution, err := client.GetExecution(ctx, &runpb.GetExecutionRequest{Name: b.cdExecution})
+	if err != nil {
+		return err
+	}
+
+	for _, condition := range execution.GetConditions() {
+		if condition.GetType() == "Completed" {
+			if condition.GetState() == runpb.Condition_CONDITION_SUCCEEDED ||
+				condition.GetState() == runpb.Condition_CONDITION_FAILED {
+				if execution.GetSucceededCount() > 0 { // should be only one CD service
+					return io.EOF
+				} else {
```

Comment:
this seems unrelated to the condition, so we can move this `if` outside of the `for`?